BankApplication/src/com/luxoft/bankapp/tests/TestBankService.java

```diff
@@ -0,0 +1,181 @@
+	private Account savingAccount = new SavingAccount(300);
+	private Account checkingAccount = new CheckingAccount(300, client1);
+	private Account[] accounts= {savingAccount, checkingAccount};
+
+	//bankService.setActiveAccount(client1,checkingAccount);
+	
+	@Test
+	public void getClientAccounts()
+	{
+		bankService.addAccount(client1,savingAccount);
+		bankService.addAccount(client1,checkingAccount);
+
+		Set<Account> list = client1.getAccounts();
+		assertTrue(list.containsAll(Arrays.asList(accounts)));
+	}
+	
+	@Test
+	public void setActiveAccount(){
+		bankService.addAccount(client1,savingAccount);
+		bankService.setActiveAccount(client1,savingAccount);
+		assertEquals(savingAccount,client1.getActiveAccount());
+	}
+	
+	@Test
+	public void getClientFromBank()
+	{
+		try {
+			bankService.addClient(bank, client1);
+		} catch (ClientExistsException e) {
+			e.printStackTrace();
+		}
+		Client client2 = bankService.getClient(bank, client1.getName());
+		assertEquals(client1, client2);
+		
+	}
+	
+	@Test
+	public void withdrawSuccesful()
+	{
+		bankService.addAccount(client1,savingAccount);
+		bankService.setActiveAccount(client1,savingAccount);
+		float balanceBefore = client1.getBalance();
+		bankService.withdraw(client1, 200); 
+		Assert.assertEquals( balanceBefore-200, client1.getBalance()); 
+	}
+	
+	@Test
+	public void depositSuccesful()
+	{
+		bankService.addAccount(client1,savingAccount);
+		bankService.setActiveAccount(client1,savingAccount);
+		float balanceBefore = client1.getBalance();
+		bankService.deposit(client1, 500); 
+		Assert.assertEquals( balanceBefore+500, client1.getBalance()); 
+	}
+	
+	@Test
+	public void testObjectSerialization(){
+		
+		bankService.addAccount(client1,savingAccount);
+		bankService.addAccount(client1,checkingAccount);
+		bankService.setActiveAccount(client1,checkingAccount);
+	
+		bankService.saveClient(client1);
+		
+		Client clientFromFile = bankService.loadClient();
+		
+		assertEquals(clientFromFile.getName(), client1.getName());
+		assertEquals(clientFromFile.getClientGender(), client1.getClientGender());
+		assertEquals(clientFromFile.getCity(), client1.getCity());
+		
+	}
+	
+	
+	@Ignore ("Not working")
+	@Test(expected=NotEnoughFundsException.class)
+	public void transferTooBigAmountFromCheckingAccount() 
+	{
+
+		bankService.addAccount(client1,savingAccount);
+		bankService.setActiveAccount(client1,savingAccount);
+		
+		Client client2 = new Client("Ben",1700,Gender.MALE, "Krakow");
+
+
+		Account savingAccount2 = new SavingAccount(300);
+		bankService.addAccount(client2,savingAccount2);
+		bankService.setActiveAccount(client2,savingAccount2);
+		
+		try {
+			bankService.addClient(bank, client1);
+			bankService.addClient(bank, client2);
+		} catch (ClientExistsException e) {
+			e.printStackTrace();
+		}
+	
+		bankService.transfer(client1, client2, 11300);
+
+
+	}
+	
+
+	@Ignore ("Not working")
+	@Test(expected=OverDraftLimitExceededException.class)
```

Comment:
Technically transfer method does not propagate this exception. It processes the exception in try-catcch block inside the method, actually prints it out. Thsat is why you see the exception in logs but test does not see it, beacause the exception is handled by printing it to console. You have to test withdraw method of the account, to check that correct exception is thrown.